lib/Doctrine/ORM/Cache/DefaultQueryCache.php

```diff
@@ -112,20 +112,29 @@ public function get(QueryCacheKey $key, ResultSetMapping $rsm, array $hints = ar
         $regionName  = $region->getName();
 
         $cm = $this->em->getClassMetadata($entityName);
+
+        $entityCacheKeys = [];
+        foreach ($entry->result as $index => $concreteEntry) {
+            $entityCacheKeys[$index] = new EntityCacheKey($cm->rootEntityName, $concreteEntry['identifier']);
+        }
+        $entries = $region->getMultiple(new CollectionCacheEntry($entityCacheKeys));
```

Comment:
Just checked...
In order to do that we would need make  `CollectionCacheEntry` know about `EntityCacheKey` and now it just depends on `CacheKey` (abstract class), so it's better keep the dependency here.